### ISSUE TITLE
All pages at once

### ISSUE THREAD

**Issue by Brecht272727** [aethletic/redtube-api#1]
How can we get all pages at once with this code?

/* Search video with filter */
print_r($redtube::searchVideo([
    'search'    => 'swap',
    'category'  => 'teens',         // see: getCategories()
    'tags'      => ['czech'],       // see: getTags()
    'stars'     => [''],            // see: getStars()
    'ordering'  => 'mostviewed',    // newest, mostviewed, rating
    'period'    => 'weekly',        // only if set "ordering": weekly, monthly, alltime
    'page'      => '1',
    'thumbsize' => 'big',
]));

**Comment by aethletic** [comment]
```php 
<?php

use Aethletic\RedtubeApi\Redtube;

require_once __DIR__ . '/redtube-api/src/Redtube.php';

$redtube = new Redtube;

$videos = [];
$pages = 2;

for ($page = 1; $page <= $pages; $page++) {
    $response = $redtube::searchVideo([
        'search'    => 'sleeping son',
        'tags'      => ['Teen'],
        'page'      => $page,
        'thumbsize' => 'big',
    ]);

    $videos = array_merge($videos, $response['videos']);

    $pages = $response['count'] % 20 == 0 ? $response['count'] / 20 : ceil($response['count'] / 20);

    echo "Total video count: {$response['count']}, current page {$page}, total pages {$pages}" . PHP_EOL;
}

echo "Found videos: " . count($videos);

```

Result:
```bash
aethletic@ping:~/sandbox/redtube$ php redtube.php
Total video count: 55, current page 1, total pages 3
Total video count: 55, current page 2, total pages 3
Total video count: 55, current page 3, total pages 3
Found videos: 55
```

`$response['count']` how many videos by request were found.
We know that on 1 page we have 20 videos.
Then `$response['count'] % 20 == 0 ? $response['count'] / 20 : ceil($response['count'] / 20)` will return the number of pages, then loop through all the pages.

I don't know for sure if can get banned for frequent requests, so be careful, if you have many pages, use delay like `sleep(1)`.

**Comment by Brecht272727** [comment]
Thanks for quick support. I will try this and let you know

**Comment by Brecht272727** [comment]
If i use only category there is something wrong. The count per category is the same, always 29068

Example:

    $response = $redtube::searchVideo([
        'category'    => 'milf',
        'page'      => $page,
    ]);

**Comment by aethletic** [comment]
It's very strange, but it looks like the `category` parameter is not working.
It always returns the same quantity.

You can check raw call `https://api.redtube.com/?data=redtube.Videos.searchVideos&output=json&category=MILF`

**Comment by Brecht272727** [comment]
Yes, i have send an email to support at hubtrafficto look at this issue.
Another question is why i have 300 items in the array (print_r($videos)) and then the array starts again?

My code:

$videos = [];
$pages = 2;

for ($page = 1; $page <= $pages; $page++) {
    $response = $redtube::searchVideo([
        //'search'    => 'milf',
        'tags'      => ['uk milf'],
        'page'      => $page,
        'thumbsize' => 'big'
    ]);

    $videos = array_merge($videos, $response['videos']);

    $pages = $response['count'] % 20 == 0 ? $response['count'] / 20 : ceil($response['count'] / 20);

    //echo "Total video count: {$response['count']}, current page {$page}, total pages {$pages}" . PHP_EOL;
       
    echo '<pre>';
    print_r($videos);
    echo '</pre>';    
       
    sleep(1);
}

echo "<br><br>Found videos: " . count($videos);

**Comment by Brecht272727** [comment]
Sometimes the array start again and again after each other. If you search on tag 'piercings' you will see that the array is starting 4 a 5 times after each other. What is going wrong here?

**Comment by Brecht272727** [comment]
> It's very strange, but it looks like the `category` parameter is not working.
> It always returns the same quantity.
> 
> You can check raw call `https://api.redtube.com/?data=redtube.Videos.searchVideos&output=json&category=MILF`

I contacted hubtraffic support and they fixed the API. Now it is working for categories also.
